### PR TITLE
support WindowEvent::DroppedFile

### PR DESCRIPTION
**Pull request by jm-observer**

### CODE REVIEW

File: src/window_handle.rs [hunk 1]
```diff
@@ -390,7 +394,17 @@ impl WindowHandle {
...
+        self.dropper_file = Some(path.clone());
+        self.event(Event::DroppedFile(DroppedFileEvent { path }));
+    }
+
     pub(crate) fn pointer_move(&mut self, pos: Point) {
+        if let Some(path) = self.dropper_file.take() {
+            self.event(Event::DroppedFileWithPosition(
+                DroppedFileWithPositionEvent { path, pos },
+            ));
+        }
```
Comment (dzhou121) [inline src/window_handle.rs:407]: The cursor position is stored in `self.cursor_position`. So we should be able to send the dropped file event with the cursor position when the dropped file event was trigger from winit.

Comment (jm-observer) [reply src/window_handle.rs:407]: I tried. It's not accurate. The mouse coordinates are not updated when you drag and drop

Comment (dzhou121) [reply src/window_handle.rs:407]: Yeah it's something at winit's end. It should be fixed when this is implemented https://github.com/rust-windowing/winit/issues/3833

Does it always emit pointer move after the file was dropped? If so, we don't need the drop file event without mouse position.

Comment (jm-observer) [reply src/window_handle.rs:407]: yes, my observation is this.

Comment (dzhou121) [reply src/window_handle.rs]: Shall we just have one event `DroppedFile` with both path and pos please?

Comment (jm-observer) [reply src/window_handle.rs]: Well, I just kept DroppedFileWithPositionEvent

**Comment by dzhou121** [comment]
rename `DroppedFileWithPosition` to `DroppedFile` maybe?

**Pull request merged**